src/Functions/array/arrayCount.cpp

```diff
@@ -81,6 +81,27 @@ using FunctionArrayCount = FunctionArrayMapped<ArrayCountImpl, NameArrayCount>;
 
 REGISTER_FUNCTION(ArrayCount)
 {
+    FunctionDocumentation::Description description = R"(
+Returns the number of elements for which `func(arr1[i], ..., arrN[i])` returns something other than `0`.
+If `func` is not specified, it returns the number of non-zero elements in the array.
+
+:::note
```

Comment:
1. I don't think we need an extra `:::note` tag here.

2. Let's just say 'array count is a higher-order-function` (without "Note that").